src/ddmd/func.d

```diff
@@ -3451,13 +3451,20 @@ extern (C++) class FuncDeclaration : Declaration
             {
                 Dsymbol st = tret.toDsymbol(null);
                 //printf("\t\treturning class/struct %s\n", tret.toChars());
-                for (Dsymbol s = st.parent; s; s = s.parent)
+
+                auto ad = st.isAggregateDeclaration();
+                assert(ad !is null);
+                // A closure is only needed when returning a _non-static_ struct/class.
+                if ((ad.storage_class & STCstatic) == 0)
                 {
-                    //printf("\t\t\tparent = %s %s\n", s.kind(), s.toChars());
-                    if (s == this)
+                    for (Dsymbol s = st.parent; s; s = s.parent)
                     {
-                        //printf("\t\treturning local %s\n", st.toChars());
-                        goto Lyes;
+                        //printf("\t\t\tparent = %s %s\n", s.kind(), s.toChars());
+                        if (s == this)
+                        {
+                            //printf("\t\treturning local %s\n", st.toChars());
+                            goto Lyes;
```

Comment:
To be frank, I do not have enough trust in DMD's testing to go that route. I am dealing with too many regressions already to risk adding one more. And I guess from the coverage tester we can already deduct that removing case 5 will not result in any changes for the code being tested.
Let's just fix the case that we are certain about (static structs/classes). I can remove the extra test (because it is testing that we need a closure for something that doesn't need a closure), add a TODO comment, add an enhancement bug issue, and perhaps a forum post. So that people can review things later and can come up with a way to break things before it actually breaks things. How about that?